Microsoft.Toolkit.Uwp.UI.Controls/InAppNotification/InAppNotification.cs

```diff
@@ -0,0 +1,133 @@
+// ******************************************************************
+
+using System;
+using System.Threading.Tasks;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// In App Notification defines a control to show local notification in the app.
+    /// </summary>
+    [TemplateVisualState(Name = StateContentVisible, GroupName = GroupContent)]
+    [TemplateVisualState(Name = StateContentCollapsed, GroupName = GroupContent)]
+    [TemplatePart(Name = DismissButtonPart, Type = typeof(Button))]
+    public sealed partial class InAppNotification : ContentControl
+    {
+        private Button _dismissButton;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="InAppNotification"/> class.
+        /// </summary>
+        public InAppNotification()
+        {
+            DefaultStyleKey = typeof(InAppNotification);
+        }
+
+        protected override void OnApplyTemplate()
+        {
+            if (_dismissButton != null)
+            {
+                _dismissButton.Click -= DismissButton_Click;
+            }
+
+            _dismissButton = (Button)GetTemplateChild(DismissButtonPart);
+
+            if (_dismissButton != null)
+            {
+                _dismissButton.Visibility = ShowDismissButton ? Visibility.Visible : Visibility.Collapsed;
+                _dismissButton.Click += DismissButton_Click;
+            }
+
+            if (Visibility == Visibility.Visible)
+            {
+                VisualStateManager.GoToState(this, StateContentVisible, true);
+            }
+            else
+            {
+                VisualStateManager.GoToState(this, StateContentCollapsed, true);
+            }
+
+            base.OnApplyTemplate();
+        }
+
+        /// <summary>
+        /// Show notification using the current template
+        /// </summary>
+        /// <param name="duration">Displayed duration of the notification in ms (less or equal 0 means infinite duration)</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public async Task ShowAsync(int duration = 0)
+        {
+            Visibility = Visibility.Visible;
+            VisualStateManager.GoToState(this, StateContentVisible, true);
+
+            if (duration > 0)
+            {
+                await Task.Delay(duration);
+                Dismiss();
+            }
+        }
+
+        /// <summary>
+        /// Show notification using text as the content of the notification
+        /// </summary>
+        /// <param name="text">Text used as the content of the notification</param>
+        /// <param name="duration">Displayed duration of the notification in ms (less or equal 0 means infinite duration)</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public async Task ShowAsync(string text, int duration = 0)
```

Comment:
This function could be done this way 
`public Task ShowAsync(string text, int duration = 0)
        {
            ContentTemplate = null;
            Content = text;
            return ShowAsync(duration);
        }`
It's better because you're giving more control to the caller when to await instead of forcing the await in your function.
also this is applied to the rest of ShowAsync